src/components/annotations/draw.js

```diff
@@ -579,21 +579,21 @@ function drawOne(gd, index, opt, value) {
                         ann.call(Lib.setTranslate, xcenter, ycenter);
 
                         update[annbase + '.x'] = xa ?
-                            (options.x + dx / xa._m) :
+                            xa.l2r(xa.p2l(xa.l2p(xa.r2l(options.x)) + dx)) :
```

Comment:
maybe I'll just make `ax.r2p` and `ax.p2r`, which I just omitted because I didn't have a use case for it for a while, but now there are a couple of places that use it.